lib/apiv2/policy.go

```diff
@@ -0,0 +1,195 @@
+type EntityRule struct {
+	// Tag is an optional field that restricts the rule to only apply to traffic that
+	// originates from (or terminates at) endpoints that have profiles with the given tag
+	// in them.
+	Tag string `json:"tag,omitempty" validate:"omitempty,tag"`
+	// Nets is an optional field that restricts the rule to only apply to traffic that
+	// originates from (or terminates at) IP addresses in any of the given subnets.
+	Nets []string `json:"nets,omitempty" validate:"omitempty,dive,cidr"`
+	// Selector is an optional field that contains a selector expression (see Policy for
+	// sample syntax).  Only traffic that originates from (terminates at) endpoints matching
+	// the selector will be matched.
+	//
+	// Note that: in addition to the negated version of the Selector (see NotSelector below), the
+	// selector expression syntax itself supports negation.  The two types of negation are subtly
+	// different. One negates the set of matched endpoints, the other negates the whole match:
+	//
+	//	Selector = "!has(my_label)" matches packets that are from other Calico-controlled
+	// 	endpoints that do not have the label “my_label”.
+	//
+	// 	NotSelector = "has(my_label)" matches packets that are not from Calico-controlled
+	// 	endpoints that do have the label “my_label”.
+	//
+	// The effect is that the latter will accept packets from non-Calico sources whereas the
+	// former is limited to packets from Calico-controlled endpoints.
+	Selector string `json:"selector,omitempty" validate:"omitempty,selector"`
+	// Ports is an optional field that restricts the rule to only apply to traffic that has a
+	// source (destination) port that matches one of these ranges/values. This value is a
+	// list of integers or strings that represent ranges of ports.
+	//
+	// Since only some protocols have ports, if any ports are specified it requires the
+	// Protocol match in the Rule to be set to "tcp" or "udp".
+	Ports []numorstring.Port `json:"ports,omitempty" validate:"omitempty,dive"`
+	// NotTag is the negated version of the Tag field.
+	NotTag string `json:"notTag,omitempty" validate:"omitempty,tag"`
+	// NotNets is an optional field that restricts the rule to only apply to traffic that
+	// does not originate from (or terminate at) an IP address in any of the given subnets.
+	NotNets []string `json:"notNets,omitempty" validate:"omitempty,dive,cidr"`
+	// NotSelector is the negated version of the Selector field.  See Selector field for
+	// subtleties with negated selectors.
+	NotSelector string `json:"notSelector,omitempty" validate:"omitempty,selector"`
+	// NotPorts is the negated version of the Ports field.
+	// Since only some protocols have ports, if any ports are specified it requires the
+	// Protocol match in the Rule to be set to "tcp" or "udp".
+	NotPorts []numorstring.Port `json:"notPorts,omitempty" validate:"omitempty,dive"`
+}
+
+type Action string
+
+const (
+	Allow Mode = "Allow"
```

Comment:
why is this `Mode` here? Shouldn't it be `PolicyType`?

Reply:
That should be `Action` not `Mode`.  I will fix.That should be `Action` not `Mode`.  I will fix.

Reply:
yes yes I meant Action